experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,269 @@
+        } else if let Some(part) = part.strip_prefix("nano") {
+            (-9, part)
+        } else if let Some(part) = part.strip_prefix("pico") {
+            (-12, part)
+        } else if let Some(part) = part.strip_prefix("femto") {
+            (-15, part)
+        } else if let Some(part) = part.strip_prefix("atto") {
+            (-18, part)
+        } else if let Some(part) = part.strip_prefix("zepto") {
+            (-21, part)
+        } else if let Some(part) = part.strip_prefix("yocto") {
+            (-24, part)
+        } else if let Some(part) = part.strip_prefix("ronto") {
+            (-27, part)
+        } else if let Some(part) = part.strip_prefix("quecto") {
+            (-30, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    // TODO: consider returning Option<(i8, &str)> instead of (0, part) for the case when the prefix is not found.
+    // TODO: consider using a trie for the prefixes.
+    // TODO: complete all the cases for the prefixes.
+    /// Get the SI prefix for base 2.
+    /// NOTE:
+    ///     if the prefix is found, the function will return (power, part without the prefix).
+    ///     if the prefix is not found, the function will return (0, part).
+    fn get_si_prefix_base_two(part: &str) -> (i8, &str) {
+        if let Some(part) = part.strip_prefix("kibi") {
+            (10, part)
+        } else if let Some(part) = part.strip_prefix("mebi") {
+            (20, part)
+        } else if let Some(part) = part.strip_prefix("gibi") {
+            (30, part)
+        } else if let Some(part) = part.strip_prefix("tebi") {
+            (40, part)
+        } else if let Some(part) = part.strip_prefix("pebi") {
+            (50, part)
+        } else if let Some(part) = part.strip_prefix("exbi") {
+            (60, part)
+        } else if let Some(part) = part.strip_prefix("zebi") {
+            (70, part)
+        } else if let Some(part) = part.strip_prefix("yobi") {
+            (80, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    // TODO: consider using a sufficient trie search for finding the unit id.
+    /// Get the unit id.
+    /// NOTE:
+    ///    if the unit id is found, the function will return (unit id, part without the unit id and without `-` at the beginning of the remaining part if it exists).
+    ///    if the unit id is not found, the function will return None.
+    fn get_unit_id<'data>(
+        part: &'data str,
+        trie: &ZeroTrie<ZeroVec<'data, u8>>,
+    ) -> Option<(usize, &'data str)> {
+        // TODO(#4379): this is inefficient way to search for an item in a trie.
+        // we must implement a way to search for a prefix in a trie.
+        let mut result = None;
+        for (index, _) in part.char_indices() {
+            let identifier = &part[..=index];
+            if let Some(value) = trie.get(identifier.as_bytes()) {
+                result = Some((value, &part[identifier.len()..]));
+            }
+        }
```

Comment:
I know it is not efficient, because of this, @sffc is implementing this PR. https://github.com/unicode-org/icu4x/pull/4383

the other error, is already cared with in the `analyze_identifier_part` fn.

However, I agree, using `split("-")` can make the search faster.